### PR TITLE
Add editorconfig file

### PR DESCRIPTION
**Pull request by tamirkamara**
The following changes are proposed:

* Use editorconfig to align on file standard (encoding, file ends, etc.), as well as some of the styling roles.
Note: this is just the ground work and a subsequent PR will align all files (didn't want to disrupt other PRs already open).
* Small fixes

### CODE REVIEW

**Review by eladiw — APPROVED** [review]
lgtm

**Comment by AsafMah** [comment]
While we are looking at the style - we do we have the using inside the namespace?
It's less convenient to work with, and it forces the ugly global:: syntax

**Comment by AsafMah** [comment]
The editorconfig file is pretty barebones, it's a good strategy to have a small one that is easier to maintain,
but another strategy is to generate a big one from the IDE and work from there.
I'm fine with both

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# CA1848: Use the LoggerMessage delegates
+dotnet_diagnostic.CA1848.severity = suggestion
+
+[*.{cs,csx,cake}]
+# 'var' preferences
+csharp_style_var_for_built_in_types = true:warning
+csharp_style_var_when_type_is_apparent = true:warning
+csharp_style_var_elsewhere = true:warning
+# others
+# csharp_style_namespace_declarations = file_scoped:warning
```
Comment (AsafMah) [inline .editorconfig:90]: Why is this commented out?

**Review by AsafMah — APPROVED** [review]
Looks good

File: .editorconfig (hunk 1 above)
Comment (tamirkamara) [reply .editorconfig:90]: It's something to uncomment after most of the PRs will be closed as this is going to change every single file.

**Pull request merged**